release-2.1: c-deps/rocksdb: fix correctness issue with snapshots and delete range

petermattis opened the pull request:
The optimization to skip sstables and swaths of keys within an sstable
covered by a range tombstone has correctness issues when used with
snapshot reads. This optimization isn't present in later versions of
RocksDB, so disable it rather than trying to fix.

Release note (bug fix): Fix a bug that could lead to data corruption or
data loss if a replica was both the source of a snapshot and was being
concurrently removed from the range. This scenario is rare, but
possible.

petermattis committed:
release-2.1: c-deps/rocksdb: fix correctness issue with snapshots and delete range

The optimization to skip sstables and swaths of keys within an sstable
covered by a range tombstone has correctness issues when used with
snapshot reads. This optimization isn't present in later versions of
RocksDB, so disable it rather than trying to fix.

Release note (bug fix): Fix a bug that could lead to data corruption or
data loss if a replica was both the source of a snapshot and was being
concurrently removed from the range. This scenario is rare, but
possible.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/48318)
<!-- Reviewable:end -->

petermattis committed:
build: do not error on warnings in RocksDB

This is a manual backport of a setting applied in later releases.

Release note: None

petermattis committed:
c-deps: bump the rebuild counter for all cmake dependencies

This is a manual backport of #46796. It makes all of the c-deps rebuild
counters different values for 2.1 vs master.

We recently bumped the version of cmake in the builder. This caused
strange build errors on CI due to switching back and forth between the
new version of cmake and the old version cmake. The makefiles created by
cmake seem to depend on the installed version of cmake in a way I don't
quite understand.

Rather than trying to add a dependency on the version of cmake, simply
bump the rebuild counter for all cmake dependencies. The one exception
is libgeos which requires the new version of cmake (and landed after the
cmake version bump was performed).

Fixes cockroachdb/dev-inf/issues/66

Release note: None

petermattis commented:
TFTR!
